### PR TITLE
Fixed failing unit tests

### PR DESCRIPTION
**Pull request by xfreebird**
Fixed the `git_commit` test
Fixed Invalid quotation in `Fastfile` test

### CODE REVIEW

File: spec/actions_specs/git_commit_spec.rb [hunk 1]
```diff
@@ -3,10 +3,10 @@
     describe "git_commit" do
       it "generates the correct git command" do
         result = Fastlane::FastFile.new.parse("lane :test do
-          git_commit(path: './README.md', message: 'message')
+          git_commit(path: './fastlane/README.md', message: 'message')
```
Comment (KrauseFx) [inline spec/actions_specs/git_commit_spec.rb]: Why?

File: lib/fastlane/fast_file.rb [hunk 2]
```diff
@@ -28,9 +28,9 @@ def parse(data)
           eval(data) # this is okay in this case
           # rubocop:enable Lint/Eval
         rescue SyntaxError => ex
-          if ex.to_s.include? "‘"
+          if data.to_s.force_encoding("utf-8").include? "\u2018"
```
Comment (KrauseFx) [inline lib/fastlane/fast_file.rb]: Why `data` instead of `ex`?

**Comment by KrauseFx** [comment]
Why do the tests not work for you? I think they work on Travis and on my machine. What error do you get?

**Comment by xfreebird** [comment]
Hi @KrauseFx,

From what I see they always fail in Travis and on my machine, see in Travis for master branch.
https://travis-ci.org/KrauseFx/fastlane/builds/81494088

And on my machine the same:

```shell
Failures:

  1) Fastlane Fastlane::FastFile git_commit generates the correct git command
     Failure/Error: end").runner.execute(:test)
     RuntimeError:
       Couldn't find file at path './README.md'
     # ./lib/fastlane/actions/git_commit.rb:27:in `block in available_options'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/config_item.rb:53:in `call'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/config_item.rb:53:in `valid?'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/config_item.rb:38:in `verify!'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/configuration.rb:50:in `block in verify_value_exists'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/configuration.rb:45:in `each'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/configuration.rb:45:in `verify_value_exists'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/configuration.rb:30:in `initialize'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/configuration.rb:18:in `new'
     # /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.17.1/lib/fastlane_core/configuration/configuration.rb:18:in `create'
     # ./lib/fastlane/configuration_helper.rb:8:in `parse'
     # ./lib/fastlane/runner.rb:134:in `block (2 levels) in execute_action'
     # ./lib/fastlane/actions/actions_helper.rb:37:in `execute_action'
     # ./lib/fastlane/runner.rb:128:in `block in execute_action'
     # ./lib/fastlane/runner.rb:127:in `chdir'
     # ./lib/fastlane/runner.rb:127:in `execute_action'
     # ./lib/fastlane/fast_file.rb:134:in `method_missing'
     # (eval):2:in `block (2 levels) in parse'
     # ./lib/fastlane/lane.rb:36:in `call'
     # ./lib/fastlane/lane.rb:36:in `call'
     # ./lib/fastlane/runner.rb:51:in `block in execute'
     # ./lib/fastlane/runner.rb:46:in `chdir'
     # ./lib/fastlane/runner.rb:46:in `execute'
     # ./spec/actions_specs/git_commit_spec.rb:7:in `block (4 levels) in <top (required)>'

  2) Fastlane Fastlane::FastFile Different Fastfiles properly shows an error message when the user uses invalid quotation
     Failure/Error: expect do
       expected Exception with "\e[31mInvalid quotation: You used the invalid quote ‘ instead of '. Make sure to use a good text editor like Sublime Text to edit your Fastfile\e[0m", got #<Encoding::CompatibilityError: incompatible character encodings: ASCII-8BIT and UTF-8> with backtrace:
         # ./lib/fastlane/fast_file.rb:31:in `include?'
         # ./lib/fastlane/fast_file.rb:31:in `rescue in block in parse'
         # ./lib/fastlane/fast_file.rb:26:in `block in parse'
         # ./lib/fastlane/fast_file.rb:25:in `chdir'
         # ./lib/fastlane/fast_file.rb:25:in `parse'
         # ./lib/fastlane/fast_file.rb:19:in `initialize'
         # ./spec/fast_file_spec.rb:243:in `new'
         # ./spec/fast_file_spec.rb:243:in `block (5 levels) in <top (required)>'
         # ./spec/fast_file_spec.rb:242:in `block (4 levels) in <top (required)>'
     # ./spec/fast_file_spec.rb:242:in `block (4 levels) in <top (required)>'

Finished in 3.58 seconds (files took 1.92 seconds to load)
526 examples, 2 failures

```

File: lib/fastlane/fast_file.rb (hunk 2 above)
Comment (xfreebird) [inline lib/fastlane/fast_file.rb]: In my case the ex didn't include any of that char when running the test. Instead it was something like ????. That was the only way to make it to work in my case.

File: spec/actions_specs/git_commit_spec.rb (hunk 1 above)
Comment (xfreebird) [inline spec/actions_specs/git_commit_spec.rb]: When I was running this test, it was searching for README.md from a different location. And it couldn't find it. I checked that location and saw that README.md was located ./fastlane/README.md relative to the location of the running test.

**Comment by xfreebird** [comment]
Basically each fork from master now fails these two tests when run with standard OSX ruby on my machine.

**Comment by KrauseFx** [comment]
I fixed the README thingy, thanks. But I'm not sure about the quotation - maybe your terminal isn't using UTF-8?

**Comment by xfreebird** [comment]
I checked that, it is "en_GB.UTF-8"

**Comment by xfreebird** [comment]
Now I tested on a different machine, and I get the same error when running rspec with the latest changes:

```ruby
Failures:

  1) Fastlane Fastlane::FastFile Different Fastfiles properly shows an error message when the user uses invalid quotation
     Failure/Error: expect do
       expected Exception with "\e[31mInvalid quotation: You used the invalid quote ‘ instead of '. Make sure to use a good text editor like Sublime Text to edit your Fastfile\e[0m", got #<Encoding::CompatibilityError: incompatible character encodings: ASCII-8BIT and UTF-8> with backtrace:
         # ./lib/fastlane/fast_file.rb:31:in `include?'
         # ./lib/fastlane/fast_file.rb:31:in `rescue in block in parse'
         # ./lib/fastlane/fast_file.rb:26:in `block in parse'
         # ./lib/fastlane/fast_file.rb:25:in `chdir'
         # ./lib/fastlane/fast_file.rb:25:in `parse'
         # ./lib/fastlane/fast_file.rb:19:in `initialize'
         # ./spec/fast_file_spec.rb:243:in `new'
         # ./spec/fast_file_spec.rb:243:in `block (5 levels) in <top (required)>'
         # ./spec/fast_file_spec.rb:242:in `block (4 levels) in <top (required)>'
     # ./spec/fast_file_spec.rb:242:in `block (4 levels) in <top (required)>'

Finished in 2.25 seconds (files took 1.46 seconds to load)
544 examples, 1 failure

Failed examples:

rspec ./spec/fast_file_spec.rb:241 # Fastlane Fastlane::FastFile Different Fastfiles properly shows an error message when the user uses invalid quotation
```

**Comment by xfreebird** [comment]
This happens only if I'm using the default ruby ```ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]```

I installed ```ruby 2.0.0p643 (2015-02-25 revision 49749) [x86_64-darwin14.1.0]``` and ```ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-darwin14]``` and the this test has passed.

I'll close this issue, since is related only to ruby version.

Thanks for your help ! :+1:

**Pull request closed without merging**